Jenkinsfile

```diff
@@ -183,15 +183,20 @@ try {
         stage('Parallel test run') {
             set_gh_status_pending(is_pr, 'Testing')
             timestamps {
-                parallel test_runs
+                try {
+                    parallel test_runs
+                } catch (Exception e) {
+                    currentBuild.result = 'UNSTABLE'
+                }
             }
         }
     } // if testinfo_sumz
 
 } catch (Exception e) {
     echo "Error: ${e}"
-    // Set currentBuild.result as FAILURE if there is an error at any point
-    currentBuild.result = 'FAILURE'
+    if (currentBuild.result != 'UNSTABLE')
+        // Set currentBuild.result as FAILURE if there is an error in building
```

Comment:
I'm having difficulties to follow the logic here. What possible values the result can have in here? If it's SUCCESS, what it means it becomes FAILURE? If it's FAILURE, why bother resetting it.